control-plane/csi-controller/src/controller.rs

```diff
@@ -233,10 +242,15 @@ impl rpc::csi::controller_server::Controller for CsiControllerSvc {
             }
         };
 
+        // Check filesystem type.
+        if !valid_fs_type(args.parameters.get("fsType")) {
+            return Err(Status::invalid_argument("Invalid filesystem type"));
+        }
+
         // Check storage protocol.
-        let protocol = match args.parameters.get("protocol") {
-            Some(p) => p.to_string(),
-            None => return Err(Status::invalid_argument("Missing storage protocol")),
+        let protocol = match parse_protocol(args.parameters.get("protocol")) {
```

Comment:
why not use the actual type rather than converting to string?

Reply:
The code below thats not changed uses this uses string comparison. Line 258. But yeah we can change it there as well.

Reply:
Yes, we could remove this as well, as it's not used anywhere else:

```
const PROTO_NVMF: &str = "nvmf";
```

Reply:
Yes, I will remove this.